File: CS/17_SmartArt/ExtractTextFromSmartArt.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using Spire.Presentation.Diagrams;
using System.IO;

namespace ExtractTextFromSmartArt
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PowerPoint document.
            Presentation presentation = new Presentation();

            //Load the file from disk.
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\ExtractTextFromSmartArt.pptx");

            //Traverse through all the slides of the PPT file and find the SmartArt shapes.
            StringBuilder st = new StringBuilder();
           st.AppendLine("Below is extracted text from SmartArt:");
            for (int i = 0; i < presentation.Slides.Count; i++)
            {
                for (int j = 0; j < presentation.Slides[i].Shapes.Count; j++)
                {
                    if (presentation.Slides[i].Shapes[j] is ISmartArt)
                    {
                        ISmartArt smartArt = presentation.Slides[i].Shapes[j] as ISmartArt;

                        //Extract text from SmartArt and append to the StringBuilder object.
                        for (int k = 0; k < smartArt.Nodes.Count; k++)
                        {
                            st.AppendLine(smartArt.Nodes[k].TextFrame.Text);
                        }
                    }
                }
            }

            String result = "Result-ExtractTextFromSmartArt.txt";

            //Save to file.
            File.WriteAllText(result, st.ToString());

            //Launch the file.
            PptDocumentViewer(result);
        }

        private void PptDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/17_SmartArt/ExtractTextFromSmartArt.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports Spire.Presentation.Diagrams
Imports System.IO

Namespace ExtractTextFromSmartArt
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PowerPoint document.
			Dim presentation As New Presentation()

			'Load the file from disk.
			presentation.LoadFromFile("..\..\..\..\..\..\Data\ExtractTextFromSmartArt.pptx")

			'Traverse through all the slides of the PPT file and find the SmartArt shapes.
			Dim st As New StringBuilder()
		   st.AppendLine("Below is extracted text from SmartArt:")
			For i As Integer = 0 To presentation.Slides.Count - 1
				For j As Integer = 0 To presentation.Slides(i).Shapes.Count - 1
					If TypeOf presentation.Slides(i).Shapes(j) Is ISmartArt Then
						Dim smartArt As ISmartArt = TryCast(presentation.Slides(i).Shapes(j), ISmartArt)

						'Extract text from SmartArt and append to the StringBuilder object.
						For k As Integer = 0 To smartArt.Nodes.Count - 1
							st.AppendLine(smartArt.Nodes(k).TextFrame.Text)
						Next k
					End If
				Next j
			Next i

			Dim result As String = "Result-ExtractTextFromSmartArt.txt"

			'Save to file.
			File.WriteAllText(result, st.ToString())

			'Launch the file.
			PptDocumentViewer(result)
		End Sub

		Private Sub PptDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace